lib/kitchen/document.rb

```diff
@@ -177,8 +202,7 @@ def modified_id_to_paste(original_id)
       return nil if original_id.nil?
```

Comment:
Ok, so feel free to hate me for saying this, but after you make the other changes... here's a refactoring that I'd like...

Do you see how all of this ID tracking code has nothing to do with the `Document` class?  It literally does not use any state of the `Document` class aside from `@id_cp_count`, and these are the only `Document` methods that use that hash.  

So to me, this yells "This code belongs in a different class!"  And yes, I realize that I put us down this path, but it is never too late for me to recognize my own mistakes :-)

So make a new class:

```ruby
class IdTracker
  def initialize(copy_suffix: '_copy_'); ... end
  def record_copied(id); ... end
  def record_cut(id); ... end
  def record_pasted(id); ... end
  def modified_id_to_paste(id); ... end
end
```

Remember to require it in `openstax_kitchen.rb`.

Aside: note how in this PR `record_id_pasted` and `modified_id_to_paste` are called together right next to each other -- maybe it should just be one method in this new class called `value_to_paste(id)` or similar.  Or `record_pasted` just moves inside `modified_id_to_paste`

Write a unit test for this class.  It should be easier than normal b/c you don't have to construct any XML.  You can just say `record_copied('foo')`, `expect(modified_id_to_paste('foo').to eq 'foo_copy_1'`, etc

Then in `Document`

```ruby
class Document
  attr_reader :id_tracker
  def initialize(...)
    @id_tracker = IdTracker.new
    # can get rid of the `@id_copy_suffix` b/c now it is inside `IdTracker`
  end
```

and in `ElementBase`

```ruby
def_delegators :document, :id_tracker
```

which will let you call `id_tracker.record_copied(id)`, etc.